Mathlib/RingTheory/Valuation/ValuativeRel.lean

```diff
@@ -81,14 +81,62 @@ macro_rules | `($a ≤ᵥ $b) => `(binrel% ValuativeRel.rel $a $b)
 
 namespace Valuation
 
-variable {R Γ : Type*} [CommRing R] [LinearOrderedCommMonoidWithZero Γ]
-  (v : Valuation R Γ)
+variable {R Γ Γ' : Type*} [CommRing R] [ValuativeRel R]
+  [LinearOrderedCommMonoidWithZero Γ] [LinearOrderedCommMonoidWithZero Γ']
 
 /-- We say that a valuation `v` is `Compatible` if the relation `x ≤ᵥ y`
 is equivalent to `v x ≤ x y`. -/
-class Compatible [ValuativeRel R] where
+class Compatible (v : Valuation R Γ) where
   rel_iff_le (x y : R) : x ≤ᵥ y ↔ v x ≤ v y
 
+namespace Compatible
+
+variable (v : Valuation R Γ) (v' : Valuation R Γ') [Compatible v] [Compatible v'] (x y : R)
+
+lemma rel_one_iff_le_one :
+    x ≤ᵥ 1 ↔ v x ≤ 1 := by
+  rw [← map_one v, ← Valuation.Compatible.rel_iff_le]
+
+lemma rel_zero_iff_nonpos :
+    x ≤ᵥ 0 ↔ v x ≤ 0 := by
```

Comment:
Can this have (if it doesn't already) a version with `v x = 0`?

Reply:
changed